### PR TITLE
Redo cached bundle + reconnect

### PR DESCRIPTION
**Pull request by natew**
Started from new repo and turned off formatting.

### CODE REVIEW

**Comment by natew** [comment]
Need to test if I can remove the hasChanged, but this wont work on my stack until https://github.com/steelbrain/pundle/issues/158 but I can test small use cases.

**Comment by steelbrain** [comment]
Can you merge master in?

**Comment by natew** [comment]
Did I miss bringing in `lastChunk` into state?

Edit: merged master.

File: packages/dev/src/index.js [hunk 1]
```diff
@@ -76,10 +76,22 @@ class Server {
       server.close()
     })
   }
+  async generateChunkForUrl(url: string): Promise<Object> {
+    const chunk = await this.generateChunk(url)
+    this.state.lastChunk[url] = chunk
```
Comment (steelbrain) [inline packages/dev/src/index.js:82]: Last chunk should be defined as an array [here](https://github.com/natew/pundle/blob/dc05298d070e36cc5b9fccdcfb228e1419a49b88/packages/dev/src/index.js#L36) and should be added to type [here](https://github.com/natew/pundle/blob/dc05298d070e36cc5b9fccdcfb228e1419a49b88/packages/dev/types.js#L41)

Comment (natew) [reply packages/dev/src/index.js:82]: Added object + type

**Comment by steelbrain** [comment]
LGTM for an initial version, Thank You

Notes for self:

Do not have a global hasChanged because this way as soon as one chunk is downloaded, hasChanged goes back to false and it doesn't let any other chunk update

So make has changed a pull-way instead of a set var, store a list of changed files and on each request to a chunk, check if any of the changed files exist in that chunk and if so, rebuild it

**Pull request merged**